File: asg32/Question2.java
import java.util.*;

class StringDemo {
    public String str = null;

    public StringDemo(String str) {
        this.str = str;
    }

    public int CountSmall() {
        char Arr[] = str.toCharArray();
        int iCnt = 0;
        int iCount = 0;
        for (iCnt = 0; iCnt < Arr.length; iCnt++) {
            if (Arr[iCnt] >= 'a' && Arr[iCnt] <= 'z') {
                iCount++;
            }
        }
        return iCount;

    }
}

class Question2 {
    public static void main(String args[]) {
        Scanner sobj = new Scanner(System.in);
        String str = null;
        int iRet = 0;

        System.out.println("Enter the String : ");
        str = sobj.nextLine();

        StringDemo SDobj = new StringDemo(str);

        iRet = SDobj.CountSmall();
        System.out.println("Number of small charachter are : " + iRet);

        sobj.close();

    }
}

File: asg42/Question2.cpp
#include<iostream>
#include<stdlib.h>
#include<unistd.h>
#include<fcntl.h>
#include<string.h>

using namespace std;

int CountSmall(char FName[]){
    char Arr[100] = {'\0'};
    int fd = 0 , iRet = 0;
    int i=0,iCount=0;

    fd = open(FName, O_RDWR);

    if(fd == -1)
    {
        cout<<"Unable to open "<<FName<<" file\n";
    }
    else
    {
        cout<<FName<<" gets opened succesfully\n";

        while((iRet = read(fd,Arr,sizeof(Arr))) != 0)
        {
            for(i=0;i<iRet;i++)  
            {
                if(Arr[i]>='a' && Arr[i]<='z'){
                    iCount++;
                }
            }
        }

    }

   return iCount;
}

int main()
{
    char FileName[30];
    int iAns=0;

    cout<<"Enter the name of file that you want to open from current directory : ";
    scanf("%s",FileName);

    iAns=CountSmall(FileName);

    cout<<"Number of Small charachter in the file are : "<<iAns;


    return 0;
}






